src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
I am unsure if the Controller should own the state of what is expanded & revealed or if the ReferenceWidget should own the state. Currently the ReferenceWidget does not retain a reference to the Controller and I am unfamiliar with the lifecycle, so this implementation has the ReferenceWidget owning this state and passing it through to the controller.